docs/articles/documentation/test-api/obtaining-data-from-the-client/examples-of-using-client-functions.md

```diff
@@ -0,0 +1,173 @@
+
+The following sample demonstrates how to create conditional test logic based on the used browser. Before running this test, install *ua-parser-js* by running `npm install ua-parser-js`.
+
+```js
+import { ClientFunction } from 'testcafe';
+import uaParser from 'ua-parser-js';
+
+fixture `My fixture`
+    .page `http://example.com`;
+
+//Returns a user-agent header sent by the browser
+const getUA = ClientFunction(() => navigator.userAgent);
+
+test('My Test', async t => {
+    const ua = await getUA();
+    const browserAlias =uaParser(ua).browser.name;
+
+    //Some common actions
+
+    //Conditional logic
+    if (browserAlias === 'Chrome'){
+        //Test logic for Chrome
+        console.log('The browser is Chrome');
+    }
+    else if (browserAlias === 'Safari'){
+        //Test logic for Safari
+        console.log('The browser is Safari');
+    }
+});
+```
+
+## Injecting External Libraries into a Page from a Test
+
+One more case of using client functions is injecting external libraries like [jQuery](https://jquery.com/) into a tested page.
+
+In order to work with jQuery from a test, you first need to add jQuery code to the tested page by using the `eval` method.
+
+```js
+await t.eval(new Function(fs.readFileSync('./jquery.js').toString()));
+```
+
+After that,  you can use jQuery in the test within a client function.
+
+```js
+import { ClientFunction } from 'testcafe';
+import fs from 'fs';
+
+fixture `My fixture`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test('My test', async t => {
+    // Adds jQuery code to the page
+    await t.eval(new Function(fs.readFileSync('./jquery.min.js').toString()));
+
+    const getElementText = ClientFunction(() => {
+        // Uses the jQuery selector
+        return $('div.column.col-2').find('legend').eq(0).text();
+    });
+
+    await t.expect(getElementText()).contains('What is your primary Operating System');
+});
+```
+
+## Accessing Child Nodes in the DOM Hierarchy
+
+To find all child elements (not [nodes](https://developer.mozilla.org/en-US/docs/Web/API/Node)) of nodes in a DOM hierarchy, TestCafe provides the [Selector.child](../selecting-page-elements/selectors/functional-style-selectors.md#child) method. However, you may need to access child nodes of some node. You can do this by using a client function.
+
+Suppose that you test the *example.html* page with the following HTML code.
+
+```js
+<!DOCTYPE html>
+<html>
+    <body id="testedpage">
+        This is my tested page. <!--This is the first child node of <body>-->
+        <p>My first paragraph.</p>
+        <p>My second paragraph.</p>
+    </body>
+</html>
+```
+
+Also suppose that you need to obtain the text *This is my tested page*. This text is considered as a child node, not a child element of the body element. So, you can’t access this text by using `Selector.child`, but can obtain the [childNodes](https://developer.mozilla.org/en-US/docs/Web/API/Node/childNodes) property from the client side.
```

Comment:
I'm not sure that `but can` is grammatically correct. Please ask the correctors.

Reply:
> This text is considered as a child node

Do you think `as` is in place here?